src/pages/api/recipients/[recId]/supply-requests.ts

```diff
@@ -79,8 +89,62 @@ async function handler(
       } catch (e) {
         return serverErrorHandler(e, res);
       }
+    case 'POST':
+      try {
+        const prisma = new PrismaClient();
+        const recipientAddSupply = await prisma.recipient.findUnique({
+          where: {
+            id: Number(recId),
+          },
+        });
+
+        if (!recipientAddSupply) {
+          return res.status(400).json({
+            error: true,
+            message: `No recipient found for id: ${recId}`,
+          });
+        }
+
+        if (!isValidRecipient) {
+          return res.status(400).json({
+            error: true,
+            message: `Please login as an authorized recipient user to access this resource`,
+          });
+        }
+        const newSupplyRequest = req.body as Prisma.SupplyRequestCreateInput;
```

Comment:
I don't think you should rely on the Prisma generated type here. It isn't clear what the input to this endpoint should be (referring in particular to how `items` should be structured) and in addition the way you manipulate `.items` doesn't match the type either.

Some other endpoints use the Prisma generated type directly as the input type but this endpoint is a bit different because not all fields will be included in the endpoint input.

Make a `NewSupplyRequestInputBody` type instead.

Reply:
On this issue, I created a newType in this file. However, the prisma-validation function I created uses this Prisma.SupplyRequestCreateInput type. I created a new type but that led to errors as well because when I imported that type into prisma-validation it led to a cycle import chain. Any suggestions on how to solve?

Reply:
I would make a separate file for types, call it `api-types.ts` or something, and add that new type there and then import it from both files